tests/src/common/test_case.rs

```diff
@@ -246,32 +268,47 @@ where
 
         self.app.update_block(next_block);
 
+        self.message_receiver.assert_empty();
+
         self
     }
 
     pub fn init_treasury(&mut self) -> &mut Self {
         self.treasury_addr = Some(TreasuryWrapper::default().instantiate::<Lpn>(&mut self.app));
+
         self.app.update_block(next_block);
 
+        self.message_receiver.assert_empty();
+
         self
     }
 
     pub fn init_profit(&mut self, cadence_hours: u16) -> &mut Self {
-        self.profit_addr = Some(ProfitWrapper::default().instantiate(
-            &mut self.app,
-            cadence_hours,
-            self.treasury_addr.as_ref().unwrap(),
-            self.timealarms.as_ref().unwrap(),
-        ));
+        const CONNECTION_ID: &str = "dex-connection";
+
+        let _: &Addr = self
+            .profit_addr
+            .insert(ProfitWrapper::default().instantiate(
+                &mut self.app,
+                cadence_hours,
+                self.treasury_addr.clone().unwrap(),
+                self.timealarms.clone().unwrap(),
+            ));
```

Comment:
why do we use `insert`? why not assign a value as it was?

Reply:
My bad. That's a leftover from the split from buy-back.